[canary] Only allow one to run at once

gregnazario opened the pull request:
This is a hack that needs to be tested against the real bors, but it should keep from sending a canary twice.

Example of it working:
https://github.com/aptos-labs/aptos-core/pull/356

Resolves https://github.com/aptos-labs/bors/issues/6

gregnazario committed:
[canary] Only allow one to run at once

zekun000 commented on bors/src/project_board.rs:
```diff
@@ -261,7 +261,41 @@ impl ProjectBoard {
+    ) -> Result<Vec<ProjectCard>> {
+        let column_id = self.canary_column.id;
+        let github = ctx.github();
+        let mut list_options = ListProjectCardsOptions {
```
this function seems just 
```
Self::list_cards(&github, self.canary_column.id)
```

gregnazario replied:
Yeah... I felt like I had that, and for some reason it didn't work, but now it does 🤷🏻‍♂️

zekun000 commented on bors/src/command.rs:
```diff
@@ -350,7 +350,25 @@ impl Command {
-            Status::InReview => ctx.pr_mut().canary_requested = true,
+            Status::InReview => {
+                let canary_running = if let Some(board) = ctx.project_board() {
+                    if let Ok(cards) = board.list_canary_cards(&ctx).await {
```
.map_or(false, |cards| cards.len() > 0)

gregnazario committed:
[signature] Fix signature code to continue to work

The latest changes to the webhook signature code was broken, so this changes it back.
